src/components/Header.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {ReactElement} from 'react';
+import {View, TextStyle} from 'react-native';
+import styles from '../styles/styles';
+import Text from './Text';
+import EnvironmentBadge from './EnvironmentBadge';
+
+type HeaderProps = {
+    /** Title of the Header */
+    title?: string | ReactElement;
+
+    /** Subtitle of the header */
+    subtitle?: string | ReactElement;
+
+    /** Should we show the environment badge (dev/stg)?  */
+    shouldShowEnvironmentBadge?: boolean;
+
+    /** Additional text styles */
+    textStyles?: TextStyle[];
+};
+
+function Header({title = '', subtitle = '', textStyles = [], shouldShowEnvironmentBadge = false}: HeaderProps) {
+    return (
+        <View style={[styles.flex1, styles.flexRow]}>
+            <View style={styles.mw100}>
+                {typeof title === 'string'
+                    ? Boolean(title) && (
+                          <Text
+                              numberOfLines={2}
+                              style={[styles.headerText, styles.textLarge, ...textStyles]}
+                          >
+                              {title}
+                          </Text>
+                      )
+                    : title}
+                {/* If there's no subtitle then display a fragment to avoid an empty space which moves the main title */}
+                {typeof subtitle === 'string'
+                    ? Boolean(subtitle) && (
```

Comment:
I think we should avoid nesting conditions. maybe ` Boolean(subtitle) && typeof subtitle === 'string' ? this : that`? We now that in negative condition `suvbtitle` can be a React node or undefined, so returning that is safe
Also `title` above can be handled that way

Reply:
@barttom With your logic if the `subtitle` is an empty string it gives the error  `Unexpected text node: . A text node cannot be a child of a <View>.`, because an empty string goes to a negative condition. Even if I remove an empty string from default values in this component it looks like it can be passed from other places as well.

Reply:
Is it a linter error? Still not sure if we should nest conditions, but if You don't have a better solution for that, can be like right now :)

Reply:
@barttom maybe another way to handle it is something like this, what do you think?
```
{isValidElement(subtitle) && subtitle}
{Boolean(subtitle) && typeof subtitle === 'string' && (
    <Text
        style={[styles.mutedTextLabel, styles.pre]}
        numberOfLines={1}
    >
        {subtitle}
    </Text>
)}
```

Reply:
I think is little bit confusing to have two separated conditions  for displaying the same variable

Reply:
@barttom Ok, then I'll leave it as is

Reply:
sure :) Thanks for trying 👍